tensorboard/components/tf_ng_tensorboard/header/containers/header.component.ts

```diff
@@ -45,15 +56,23 @@ const selectActivePluginIndex = createSelector(
   styleUrls: ['./header.component.css'],
 })
 export class HeaderComponent {
-  activePluginIndex$ = this.store.pipe(select(selectActivePluginIndex));
-  plugins$ = this.store.pipe(select(selectPlugins));
+  readonly activePlugin$ = this.store.pipe(select(getActivePlugin));
```

Comment:
q: Why do we need to `select` over `getActivePlugin` here? It’s already
a selector, right?

Reply:
Below is my understanding of things:
- Selector: maps State into smaller state (you can create intermediate UI-only state with multiple data)
- `select`: Create a function that returns an observable that invokes the `Selector` underneath.

Reply:
Okay, so `createSelector` is just function composition with memoization,
like (simplified)

```js
function createSelector(dep1, dep2, dep3, fn) {
  const memo = new Map();
  return function selector(store) {
    if (!memo.has(store)) {
      const val1 = dep1(store);
      const val2 = dep2(store);
      const val3 = dep3(store);
      const result = fn(val1, val2, val3);
      memo.set(store, result);
    }
    return memo.get(store);
  };
}
```

and `select` creates a function that takes… an `Observable` of `State`
or of `Store<State>`, where `Store<State>` is an `Observable<State>`
itself? and then (maybe after extracting an inner state) passes this
through an observable-aware `map` (`fmap`) to project out the substate
via the selector, resulting in an `Observable<Substate>`?

It’ll take a while for this to solidify—I don’t yet see how the
primitives fit together—and I’m not having the easiest time reading the
RxJS and ngrx sources, but it’d be helpful to know if this is
approximately on the right track.